src/lambda/math.rs

```diff
@@ -118,18 +159,52 @@ impl Eval for Math {
                 Math::Sum(exprs) => {
                     let results: Vec<_> = exprs.eval(ctx, conc).await?;
 
-                    results.into_iter().try_fold(0i64.into(), |lhs, rhs| {
-                        try_f64_operation(&lhs, &rhs, ops::Add::add)
-                            .or_else(|| try_u64_operation(&lhs, &rhs, ops::Add::add))
-                            .or_else(|| try_i64_operation(&lhs, &rhs, ops::Add::add))
-                            .ok_or(EvaluationError::ExprEvalError("sum".into()))
-                    })?
+                    try_sum_operation(&results, None)?
                 }
             })
         })
     }
 }
 
+fn format_error(error: Option<&str>, op: &str) -> String {
+    error.map(|e| format!("{}-{}", e, op)).unwrap_or(op.into())
+}
+
+fn try_mean_operation(
+    exprs: &[ConstValue],
+    error: Option<&str>,
+) -> Result<ConstValue, EvaluationError> {
+    let error = format_error(error, "mean");
+
+    try_sum_operation(exprs, Some(&error))
+        .ok()
+        .map(|sum| try_div_operation(&sum, &exprs.len().into(), Some(&error)))
+        .ok_or(EvaluationError::ExprEvalError(error))?
+}
+
+fn try_sum_operation(
+    exprs: &[ConstValue],
+    error: Option<&str>,
+) -> Result<ConstValue, EvaluationError> {
+    exprs.iter().try_fold(0i64.into(), |lhs, rhs| {
+        try_f64_operation(&lhs, rhs, ops::Add::add)
+            .or_else(|| try_u64_operation(&lhs, rhs, ops::Add::add))
+            .or_else(|| try_i64_operation(&lhs, rhs, ops::Add::add))
+            .ok_or(EvaluationError::ExprEvalError(format_error(error, "sum")))
+    })
+}
+
+fn try_div_operation(
+    lhs: &ConstValue,
+    rhs: &ConstValue,
+    error: Option<&str>,
+) -> Result<ConstValue, EvaluationError> {
+    try_f64_operation(lhs, rhs, ops::Div::div)
+        .or_else(|| try_u64_operation(lhs, rhs, ops::Div::div))
+        .or_else(|| try_i64_operation(lhs, rhs, ops::Div::div))
+        .ok_or(EvaluationError::ExprEvalError(format_error(error, "div")))
+}
```

Comment:
this was a hackish way to be able to re-use operations (because `Mean` and `Median` both uses other operations), but ideally, we should use `Math::div` or `Math::sum` itself to accomplish it. I was not successful in implementing it, but if you have any suggestions, I'd be happy to implement it.

I did not find a way to parse a `ConstValue` into an `Expression`, that's why I was unable to reuse operations directly